hostprocess/build-calico.ps1

```diff
@@ -0,0 +1,46 @@
+param(
+    [string]$repository = "sigwindowstools",
+    [version]$minCalicoVersion = "3.19.0",
+    [version]$minK8sVersion = "1.22.0"
+)
+
+pushd calico
+write-host "build calico"
+$calicoVersions = (curl -L https://api.github.com/repos/projectcalico/calico/releases | ConvertFrom-Json) | % tag_name
+foreach($calicoVersion in $calicoVersions)
```

Comment:
I have not tested with any of the other calico versions... I did modify the startup scripts to get this to work with hostprocess.  I am hesitant to add more calico versions without further investigation.

Reply:
I used `pwsh` just because have no experience to write bash scripts and don't know how to code the same script but using bash.

Reply:
I understand that we can't test all pairs `calico` and `kube-proxy` but anyway these may contain some issues and only user can decide which versions better for him. So, my proposal to allow all of them.

As additional point it makes possible to upgrade k8s cluster to any public version include minor versions.

Reply:
I have tested the last of calico on the last of kubernetes. For me it works very good.

Reply:
Thanks for doing this and testing the various versions.  

I thought about this some more and at this time I don't want to add all the different versions matrixes. As you point these are more demonstrations of how this works and users should be able to adjust them to their own needs pretty easily.  I want to avoid getting into debugging CNI configurations in this repo as we did the files in the kubeadm folder.  We don't have the breadth of CNI knowledge to maintain multiple cni's and versions.

The long-term goal for kube-proxy is move it k/k repo as an official image.  I will be working on that in the coming weeks.

For the CNI's, these should be hosted by cni providers longer term. I started to upstream the work for Calico but due to some limitations in the host-process design as of now it wasn't a good solution.  We are re-working the underlying solution so it will be easier to integrate and that should be coming in the next couple months.

That said we can maintain a few limited options in the meantime and I will try to add a job using the scripts we have currently.  Thanks again for your continued contributions and feedback. I hope you keep it coming.